html/js/display-tag.js

```diff
@@ -58,94 +114,85 @@ function fitBoundsToAllLayers(mapToUpdate) {
-      "}",
-    settings = {
-      headers: { Accept: 'application/sparql-results+json' },
-      data: { query: sparqlQuery }
-    };
-
-  $.ajax(endpointUrl, settings).then(callback);
-}
+async function getOpenStreetMapFromWikidata(id) {
+  const endpointUrl = 'https://query.wikidata.org/sparql';
+  const sparqlQuery = `SELECT ?OpenStreetMap_Relations_ID WHERE {
+    wd:${id} wdt:P402 ?OpenStreetMap_Relations_ID.
+  }`;
+  const settings = {
+    headers: { Accept: 'application/sparql-results+json' }
+  };
+
+  const response = await fetch(`${endpointUrl}?query=${encodeURIComponent(sparqlQuery)}`, settings);
+  const data = await response.json();
 
-function getOsmDataFromOverpassTurbo(id, callback) {
-  $.ajax('https://overpass-api.de/api/interpreter?data=relation%28' + id + '%29%3B%0A%28._%3B%3E%3B%29%3B%0Aout%3B').then(callback);
+  return data;
 }
 
-function getGeoJsonFromOsmRelation(id, callback) {
-  getOsmDataFromOverpassTurbo(id, function (xml) {
-    callback(osmtogeojson(xml));
-  });
+async function getGeoJsonFromOsmRelation(id) {
+  const response = await fetch(`https://polygons.openstreetmap.fr/get_geojson.py?params=0&id=${encodeURIComponent(id)}`);
+  const data = response.json();
+  const geoJsonRewind = new GeoJSONRewind();
+
+  return geoJsonRewind.rewind(data);
 }
 
 function displayPointers(pointers) {
-  runCallbackOnJson(function (actualMap) {
-    const markers = [];
-    for (const pointer of pointers) {
-      let coordinates;
-
-      // If pointer is an array, it just contains (lat, lng) geo coordinates
-      if (Array.isArray(pointer)) {
-        coordinates = pointer;
-      }
-      // Otherwise we have a structured object
-      // e.g. from a map element of a knowledge panel
-      else {
-        coordinates = [pointer.geo.lat, pointer.geo.lng];
-      }
-
-      const marker = new L.marker(coordinates);
-      markers.push(marker);
+  const map = createLeafletMap();
+  const markers = [];
+  for (const pointer of pointers) {
+    let coordinates;
+
+    // If pointer is an array, it just contains (lat, lng) geo coordinates
+    if (Array.isArray(pointer)) {
+      coordinates = pointer;
     }
-
-    if (markers.length > 0) {
-      L.featureGroup(markers).addTo(actualMap);
-      fitBoundsToAllLayers(actualMap);
-      actualMap.setZoom(8);
+    // Otherwise we have a structured object
+    // e.g. from a map element of a knowledge panel
+    else {
+      coordinates = [pointer.geo.lat, pointer.geo.lng];
     }
-  });
+
+    const marker = new L.marker(coordinates);
+    markers.push(marker);
+  }
+
+  if (markers.length > 0) {
+    L.featureGroup(markers).addTo(map);
+    fitBoundsToAllLayers(map);
+    map.setZoom(8);
+  }
 }
 
-function displayMap(pointers, wikidataObjects) {
+export async function displayMap(pointers, wikidataObjects) {
   if (pointers.length > 0) {
     displayPointers(pointers);
   }
 
   for (const wikidataObject of wikidataObjects) {
```

Comment:
I'm getting an error in the console:

display-tag.js:193 Uncaught (in promise) TypeError: t is not iterable
    at displayMap (display-tag.js:193:32)
    at HTMLDocument.<anonymous> (2000000000001:1815:9)

Reply:
Oh interesting. Which page did you use to test? Is it one that's mapped to Wikidata, but the Wikidata page doesn't have any geography assigned? http://world.openfoodfacts.localhost/country/cambodia works for me, for example. I'll add a check to ensure it doesn't try to enumerate on `undefined`/`null`